runtime/gc_structs/ValueTypesIterator.hpp

```diff
@@ -0,0 +1,74 @@
+
+/*******************************************************************************
+ * Copyright (c) 1991, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Structs
+ */
+
+#if !defined(VALUETYPESITERATOR_HPP_)
+#define VALUETYPESITERATOR_HPP_
+
+#include "j9.h"
+#include "j9cfg.h"
+#include "modron.h"
+
+/**
+ * Iterate over all MethodType slots in a class which contain an object reference.
+ * @ingroup GC_Structs
+ */
+class GC_ValueTypesIterator
+{
+protected:
+	bool _state;
+	j9object_t *_defaultValuePtr;
+	
+public:
+	GC_ValueTypesIterator(J9Class *_clazzPtr)
+	{    
+		_state = false;
+		if (J9_IS_J9CLASS_VALUETYPE(_clazzPtr) && (!J9CLASS_IS_ARRAY(_clazzPtr))){
+			_defaultValuePtr = &(_clazzPtr->flattenedClassCache[0].header.defaultValue);
```

Comment:
the `[0]` is not necesarry

Reply:
should I change all flattenedClassCache[0].header to flattenedClassCache->header? Since flattenedClassCache is a pointer and it does not allow to do flattenedClassCache.header

Reply:
@tajila